leapmotion-examples/LeapMotionCoreAssets
# Feature leap ipd

Aligns virtual cameras to Leap cameras, without modifying Oculus code.
Updates QuickSwitch to also toggle alignment.

Because Oculus uses LateUpdate to position their cameras it is necessary to override this by enforcing execution order in a non-project-specific way. This is accomplished by controlling the order in which scripts are enabled.

To test:
- Hands and world appear huge in Lightweight Passthrough (Stereo)
- Hand sizes switch in QuickSwitcher demo scene
- All of these things work in BOTH editor & builds.
- World alignment is good when oculus position tracker is working (see below)

When the oculus position tracker is working, there should be almost no drift between the real and virtual worlds. To test this, choose a passthrough scene (I recommend VRWidgets or PassthroughTouchGlow) and align a real object with a virtual one, then examine from multiple perspectives. So long as the oculus position tracking is working, the alignment should remain.
NOTE: Oculus filters its rotation tracking heavily. Consequently, the virtual objects will lag *behind* the real objects when ones head is rotating.
